### ISSUE TITLE
Low GPU usage and High loss and WER

### ISSUE THREAD

**Issue by AmeerHamzaRiaz** [mozilla/DeepSpeech#1767]
_[Please close the issue right-away if does not qualify, the reason for posting here instead of discourse is that I need quick help thanks]_ :/
Hello, I've trying to train deepspeech on Urdu (Nastaliq Script which is similar to Arabic) using [Urdu phonetically rich dataset](http://csalt.itu.edu.pk/PRUSCorpus/index.html) which is of only 70 minutes.

Obviously I'm getting very bad result these are the results of 10 epochs. How can I improve these results with such a small dataset. Also , the training takes too much time I checked my gpu utilization and it is less than 10%. while my CPU is running at around 95+ %.

**The command which i used to start training:**
`python -u DeepSpeech.py   --train_files /home/gamelab/DeepSpeech/data/CV-Compatible-Urdu-Dataset/cv-valid-train/cv-valid-train.csv   --dev_files /home/gamelab/DeepSpeech/data/CV-Compatible-Urdu-Dataset/cv-valid-dev/cv-valid-dev.csv   --test_files /home/gamelab/DeepSpeech/data/CV-Compatible-Urdu-Dataset/cv-valid-test/cv-valid-test.csv   --train_batch_size 2   --dev_batch_size 2   --test_batch_size 1   --epoch 20   --validation_step 1   --export_dir /home/gamelab/DeepSpeech/data/results/model_export/   --checkpoint_dir /home/gamelab/DeepSpeech/data/results/checkout/   --alphabet_config_path /home/gamelab/DeepSpeech/data/alphabet.txt   --lm_binary_path /home/gamelab/DeepSpeech/data/lm.binary   --lm_trie_path /home/gamelab/DeepSpeech/data/trie`

**These are my results of training.**

```
I Training epoch 0...
I Training of Epoch 0 - loss: 223.102412                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:01:55 Time:  1:01:55

I Validating epoch 0...
I Validation of Epoch 0 - loss: 188.416498                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:06:26 Time:  0:06:26

I Training epoch 1...
I Training of Epoch 1 - loss: 224.996908                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:06:11 Time:  1:06:11
I 
Validating epoch 1...
I Validation of Epoch 1 - loss: 209.471827                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:15 Time:  0:07:15

I Training epoch 2...
I Training of Epoch 2 - loss: 224.456416                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:04:52 Time:  1:04:52

I Validating epoch 2...
I Validation of Epoch 2 - loss: 209.150351                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:16 Time:  0:07:16

I Training epoch 3...
I Training of Epoch 3 - loss: 223.924234                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:04:55 Time:  1:04:55

I Validating epoch 3...
I Validation of Epoch 3 - loss: 206.500756                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:16 Time:  0:07:16

I Training epoch 4...
I Training of Epoch 4 - loss: 223.235921                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:04:55 Time:  1:04:55

I Validating epoch 4...
I Validation of Epoch 4 - loss: 207.368481                                                                                                                   d
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:16 Time:  0:07:16

I Training epoch 5...
I Training of Epoch 5 - loss: 222.731319                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:04:55 Time:  1:04:55

I Validating epoch 5...
I Validation of Epoch 5 - loss: 205.300650                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:16 Time:  0:07:16

I Training epoch 6...
I Training of Epoch 6 - loss: 222.233065                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:05:00 Time:  1:05:00

I Validating epoch 6...
I Validation of Epoch 6 - loss: 205.496103                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:15 Time:  0:07:15

I Training epoch 7...
I Training of Epoch 7 - loss: 221.451717                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:05:03 Time:  1:05:03

I Validating epoch 7...
I Validation of Epoch 7 - loss: 203.135249                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:16 Time:  0:07:16

I Training epoch 8...
I Training of Epoch 8 - loss: 214.645133                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:05:05 Time:  1:05:05

I Validating epoch 8...
I Validation of Epoch 8 - loss: 191.030465                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:15 Time:  0:07:15

I Training epoch 9...
 41% (51 of 124) |#########################################                                                            | Elapsed Time: 0:23:05 ETA:   0:33:46
I Training of Epoch 9 - loss: 189.272035                                                                                                                     
100% (124 of 124) |####################################################################################################| Elapsed Time: 1:05:17 Time:  1:05:17

I Validating epoch 9...
I Validation of Epoch 9 - loss: 157.834084                                                                                                                   
100% (36 of 36) |######################################################################################################| Elapsed Time: 0:07:19 Time:  0:07:19
```

**My GPU utilization**(Stays at mostly 0% occasionally goes to 10%)
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.130                Driver Version: 384.130                   |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:01:00.0  On |                  N/A |
| 43%   35C    P8     8W / 120W |    148MiB /  6069MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
```

**My PC SPECS ARE:**
Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
cores=4 threads=8
ram-size: 16GiB
GPU: Nvidia1060 6gb
Tensorflow version: v1.12.0-rc2-0-g748435b8ef 1.12.0-rc2
Python 3.5.2

```

Thanks , if anyone can help me out in decreasing training time and increasing accuracy, I would be grateful.

**Comment by reuben** [comment]
Make sure you install `tensorflow-gpu` and not `tensorflow`, increase the batch size, make sure each audio clip is short, under 10 seconds if possible.

**Comment by nullbyte91** [comment]
As @reuben mentioned, Make sure you have installed tensorflow-gpu. You can check by executing below commands on Python console,
>>> import tensorflow as tf
>>> sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))

**Comment by lissyx** [comment]
> _[Please close the issue right-away if does not qualify, the reason for posting here instead of discourse is that I need quick help thanks]_ :/

So we answered the same day, and there's no reaction. Closing.
